live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/LiveIngesterEventHandler.java

```diff
@@ -54,7 +58,9 @@ public void configure()
                 .transacted()
                 .routeId(ROUTE_ID)
                 .log(DEBUG, "Received repo event : ${header.JMSMessageID}")
-                .filter(exchange -> repoEventFilterHandler.filterNode(exchange, integrationProperties.alfresco().filter()))
+                .setBody(camelEventMapper::repoEventFrom)
+                .process(exchange -> repoEventFilterHandler.handle(exchange, integrationProperties.alfresco().filter()))
+                .filter(exchange -> BooleanUtils.isNotTrue(exchange.getProperty(DENY_NODE, Boolean.class)))
```

Comment:
Could we just return a boolean from `repoEventFilterHandler.handle` and use that to filter (I'm aware I'm suggesting basically the code that used to be here).

Reply:
The case I particularly didn't like with previous approach was mixing processing with filtering.
There shouldn't be more responsibilities for filter which it had in the previous version - and now it is just a simple filter.
Processors, however, should manipulate `Exchange` and the processor I added does that under some conditions (adds a property to deny event or alters event type). 
Ideally, that processor could have been split into 2: one to determine filtering results and second one to alter the event type but I thought this solution is good enough.

Reply:
The thing I don't like about this approach is that we're using global variables. There's no easy way to find where that property is being set. I agree that it's not nice to hide processing inside a filter though.

Reply:
TBH I would like to see a mixed (this x previous) approach:
`.setBody(camelEventMapper::repoEventFrom)
.filter(exchange -> repoEventFilterHandler.handle(exchange, integrationProperties.alfresco().filter())) //sets NEW_EVENT_TYPE_PROP
.process(exchange -> SecurityContextHolder.setContext(securityContext))
.process(eventProcessor::process) //processes the NEW_EVENT_TYPE_PROP and alters the RepoEvent` 
but I understand your arguments so I'm not forcing it - I'm happy when you're happy

Reply:
`Exchange` object (message) is not global - it is request specific (unfortunately in our case rather route specific as we don't pass it further but create another one). And I would treat property of `Exchange` as metadata of message which is one of basic EIP concepts. 
I think we could take out these to lines 
```
.process(exchange -> repoEventFilterHandler.handle(exchange, integrationProperties.alfresco().filter()))
.filter(exchange -> BooleanUtils.isNotTrue(exchange.getProperty(DENY_NODE, Boolean.class)))
```
and include them (their logic to be precise) in 
```
.process(eventProcessor::process)
```
That way `EventProcessor` will be responsible for everything.
There could be another way to dela with that - set `Exchange` body to some `empty` object instead of setting `Exchange` property, That would result in:
```
.filter(exchange -> !exchange.getIn().getBody().equals("empty"))
```
Any thoughts?

Reply:
I like the idea of changing the whole body. That seems easier to follow than using a property on the exchange.

Having a single `process` statement to do everything also seems easier to follow, but it wouldn't be easy to see that the message is sometimes filtered out.